napari/_qt/layer_controls/qt_image_controls.py

```diff
@@ -98,8 +98,12 @@ def __init__(self, layer):
         self.grid_layout.addWidget(self.contrastLimitsSlider, 1, 1)
         self.grid_layout.addWidget(QLabel('gamma:'), 2, 0)
         self.grid_layout.addWidget(self.gammaSlider, 2, 1)
-        self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
-        self.grid_layout.addLayout(colormap_layout, 3, 1)
+
+        # The colormap widget won't be created when image has a rgb layer
+        if layer.rgb is False:
+            self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
+            self.grid_layout.addLayout(colormap_layout, 3, 1)
```

Comment:
I'd undo these changes, I think if we are going with the hide approach it is fine to add them and then hide them.